momento-provider/src/test/java/com/google/code/ssm/providers/momento/TestPOC.java

```diff
@@ -0,0 +1,128 @@
+public class TestPOC {
+
+    private static final String MOMENTO_AUTH_TOKEN = "<YOUR MOMENTO TOKEN GOES HERE>";
+    // IMPORTANT: This presumes you have already created a cache in Momento named 'example-cache'
+    private static final String MOMENTO_CACHE_NAME = "example-cache";
+
+    @Configuration
+    public static class ExampleCacheConfiguration extends AbstractSSMConfiguration {
+        @Bean
+        @Override
+        public CacheFactory defaultMemcachedClient() {
+            final MomentoConfiguration conf = new MomentoConfiguration();
+            conf.setConsistentHashing(true);
+            conf.setDefaultTtl(300);
+            conf.setCacheName(MOMENTO_CACHE_NAME);
+            conf.setMomentoAuthToken(MOMENTO_AUTH_TOKEN);
+            final CacheFactory cacheFactory = new CacheFactory();
+            cacheFactory.setCacheClientFactory(new MomentoCacheClientFactory());
+            // Use a MomentoAddressProvider to be explicit
+            cacheFactory.setAddressProvider(new MomentoAddressProvider());
+            cacheFactory.setConfiguration(conf);
+            return cacheFactory;
+        }
+    }
+
+    @Autowired
+    private CacheFactory exampleCacheFactory;
+
+    @Data
+    @JsonSerialize
+    public static class TestObject implements Serializable {
+        int id;
+        String name;
+    }
+
+
+    @ReadThroughSingleCache(namespace = "TestString", expiration = 3600)
+    @SneakyThrows
+    public String getSimpleObject(@ParameterValueKeyProvider String complexObjectPk) {
+        Cache cache = exampleCacheFactory.getObject();
+        if (cache != null) {
+            String result = cache.get(complexObjectPk, SerializationType.PROVIDER);
+            if (result != null) {
+                return result;
+            }
+        }
+        // Otherwise, retrieve object from source location, e.g. DynamoDB, S3
+        // return getObjectFromSource(complexObjectPk);
+        return null;
+    }
+
+    @ReadThroughSingleCache(namespace = "TestObject", expiration = 3600)
+    @SneakyThrows
+    public TestObject getComplexObject(@ParameterValueKeyProvider String objectKey) {
+        Cache cache = exampleCacheFactory.getObject();
+        ObjectMapper mapper = new ObjectMapper();
+        if (cache != null) {
+            TestObject obj = mapper.readValue((String) cache.get(objectKey, SerializationType.PROVIDER), TestObject.class);
+            if (obj != null) {
+                return obj;
+            }
+        }
+        // Otherwise, retrieve object from source location, e.g. DynamoDB, S3
+        // return getObjectFromSource(objectKey);
+        return null;
+    }
+
+
+    @Test
+    @SneakyThrows
+    @Ignore // Comment out if you'd like to test this out yourself
+    public void testPocSimple() {
+        String test = "fooBar";
+        exampleCacheFactory.getObject().set("test-key", 500, test, SerializationType.PROVIDER);
+        String result = getSimpleObject("test-key");
```

Comment:
look, this _really_ seems like this test should only ever print "Nothing found." Otherwise, your getSimpleObject() and getComplexObject() cache views collide in the same namespace.

I think it's broken to have a namespace parameter that does not namespace your keys (whether via a separate cache or a key prefix or what have you).

Reply:
That's totally fair, and to be honest I need to dig more into how to utilize the `namespace` param

Reply:
https://ragnor.github.io/simple-spring-memcached/com/google/code/ssm/api/ReadThroughSingleCache.html

> public abstract [String](http://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true) namespace
> 
> A namespace that is added to the key as it is stored in the distributed cache. This allows differing object that may have the same ID to coexist. This value must be assigned.

Based on the documentation it seems that this wrapper does not correctly respect the required namespace parameter. So yeah, this test should _not work_ as written; if it gets fooBar back it's a bug.

Reply:
It's okay if you want to fix this bug in a follow up pr, but if you do then you'll want to open a ticket & resolve it straight away. It's an api facet that's exposed, but only partially implemented (user will get() ostensibly namespaced objects that are the wrong type and their app will crash through no fault of their own)

Reply:
I'm going to merge this for now so that we can hand this off, but I have a lot of concerns about this annotation to begin with. I don't see anywhere in the library how this is just "automagically" added as a key prefix and this needs to be investigated